pandas/io/tests/test_gbq.py

```diff
@@ -430,10 +426,16 @@ def test_should_properly_handle_valid_integers(self):
         df = gbq.read_gbq(query, project_id=PROJECT_ID)
         tm.assert_frame_equal(df, DataFrame({'VALID_INTEGER': [3]}))
 
+    def test_should_properly_handle_valid_longs(self):
+        query = 'SELECT 4370565731996369294 as VALID_LONG'
+        df = gbq.read_gbq(query, project_id=PROJECT_ID)
+        tm.assert_frame_equal(
+            df, DataFrame({'VALID_LONG': [4370565731996369294]}))
+
     def test_should_properly_handle_null_integers(self):
         query = 'SELECT INTEGER(NULL) as NULL_INTEGER'
         df = gbq.read_gbq(query, project_id=PROJECT_ID)
-        tm.assert_frame_equal(df, DataFrame({'NULL_INTEGER': [np.nan]}))
+        tm.assert_frame_equal(df, DataFrame({'NULL_INTEGER': [None]}))
```

Comment:
casting to float means lost precision for integers > `2^53`.
I think it's better to lost performance than lost precision.
Lost precision means trouble when unnoticed.

I started with solution which raises exception for `NULL` in `INTEGER` column. This forces user to eliminate `NULL`-s (eg. by explicitly casting to `FLOAT` at BigQuery side; there is `float()` function). But finally I think it is less convenient.

My proposed solution is: 
- to store `INTEGER` column without `NULL`-s as `dtype = int64`
- and `INTEGER` with `NULL`-s as `dtype = object`

It mimics behaviour from `read_csv`.  Similarly for `BOOLEAN` columns (bool or object).